ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.cpp

```diff
@@ -3,12 +3,62 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "ttnn/operations/reduction/generic/generic_reductions.hpp"
+#include "ttnn/operations/data_movement/pad/pad.hpp"
+#include "ttnn/operations/data_movement/slice/slice.hpp"
 #include "ttnn/operations/data_movement/transpose/transpose.hpp"
 #include "ttnn/operations/eltwise/unary/unary.hpp"
 #include "ttnn/operations/eltwise/binary/binary_composite.hpp"
 #include "ttnn/operations/reduction/generic/device/reduce_op.hpp"
 #include "ttnn/operations/core/core.hpp"
 
+// Some tensors are pre-padded with 0s. E.g. Those generated via from_torch.
+// Therefore need to always pad tensors again. To do that, convert to row major,
+// pad, and then convert back to tile layout.
+// Limitations of pad require transpose, un-transpose, and then slicing to isolate values of interest.
+// End result will be padded, and after reduce is done, will need to be sliced back.
+ttnn::Tensor pad_tensor_with_value(const ttnn::Tensor& input_tensor, float pad_value) {
```

Comment:
can't call `ttnn::pad`?

Reply:
ah, this is basically rewrite of alignment padding...

Reply:
The problem is that things like from_torch already pre-pad with 0s. That needs to be undone and padded properly. ttnn::pad isn't sufficient.

Reply:
Deferred to a future PR.